partiql-eval/src/error.rs

```diff
@@ -0,0 +1,58 @@
+use crate::eval::evaluable::Evaluable;
+use crate::eval::expr::EvalExpr;
+use crate::eval::EvalContext;
+use partiql_value::{Tuple, Value};
+use std::borrow::Cow;
+use thiserror::Error;
+
+/// All errors that occur during evaluation.
+#[derive(Debug)]
+pub struct EvalErr {
+    pub errors: Vec<EvaluationError>,
+}
+
+/// An error that can happen during evaluation.
+/// TODO: decide on splitting out logical to eval plan to separate error enum
```

Comment:
Planning and Evaluation are distinct phases with differing types of errors. They should be disjoint error types.